main.rb

```diff
@@ -0,0 +1,117 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    counter = 0
+    while counter < size
+      if instance_of(Hash) || instance_of?(Range)
+        new_hash = to_a
+        yield(new_hash[counter])
+      else
+        yield(self[counter])
+      end
+      counter += 1
+    end
+  end
+
+  def my_each_with_index
+    counter = 0
+    while counter < size
+      if instance_of(Hash) || instance_of?(Range)
+        new_hash = to_a
+        yield(new_hash[counter], counter)
+      else
+        yield(self[counter], counter)
+      end
+      counter += 1
+    end
+  end
+
+  def my_select
+    if instance_of?(Hash)
+      new_hash = {}
+      my_each { |key, value| new_hash[key] = value if yield(key, value) }
+      new_hash
+    else
+      arr = []
+      my_each { |item| arr.push(item) if yield(item) }
+      arr
+    end
+  end
+
+  def my_all?(type = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false }
+    elsif type.nil?
+      my_each { |item| return false if item.nil? }
+    else
+      my_each { |item| return false if item.class != type && item.class.superclass != type }
+    end
+    true
+  end
+
+  def my_any?(type = nil)
+    if block_given?
+      my_each { |item| return true if yield(item) == true }
+    elsif type.nil?
+      my_each { |item| return true if item.nil? }
+    else
+      my_each { |i| return true if i.instance_of?(type.class) || i.class.superclass == type || i.is_a?(type) }
+    end
+    false
+  end
+
+  def my_none?(type = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == true }
+    elsif type.nil?
+      my_each { |item| return false if item.nil? && item != false }
+    else
+      my_each { |item| return false if item.instance_of?(type) || item.class.superclass == type || item.is_a?(type) }
+    end
+    true
+  end
```

Comment:
#### my_none?
- [ ] 15. #my_none? when no block or argument is given returns true only if none of the collection members is true
- [ ] 16. #my_none? when a Regex is passed as an argument returns true if none of the collection matches the Regex
- [ ] 17. #my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex
- [ ] 18. #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
- [ ] 19. #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern